enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -0,0 +1,669 @@
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  installStrategy:
+    type: Placements
+    placements:
+    - name: global-placement
+      namespace: default
+      addonTemplate:
+        installVersion: v1
+        configs:
+        - group: addon.open-cluster-management.io
+          resource: addondeploymentconfigs
+          name: global-config
+    - name: canary-placement
+      namespace: default
+      addonTemplate:
+        installVersion: v1
+        configs:
+        - group: addon.open-cluster-management.io
+          resource: addondeploymentconfigs
+          name: canary-config
+status:
+  supportedVersions:
+  - v1
+  - v2
+  defaultVersion:
+  - v2
+```
+
+The rollbacked `ManagedClusterAddOn` would be like:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: helloworld
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-agent-addon
+  installVersion: v1
+Status:
+…
+  lastVersion: v2
+  currentVersion: unknown
+  conditions:
+  - lastTransitionTime: "2022-11-02T05:48:12Z"
+    message: Rollingback addon to version v1.
+    reason: Rollingback
+    status: "True"
+    type: Progressing
+```
+
+Then becomes 
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: helloworld
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-agent-addon
+  installVersion: v1
+Status:
+…
+  lastVersion: v2
+  currentVersion: v1
+  conditions:
+  - lastTransitionTime: "2022-11-02T05:48:12Z"
+    message: install completed with no errors.
+    reason: Succeed
+    status: "Flase"
+    type: Progressing
+```
+
+If the rollbacked version is invalid, or any other error occurs during rollback, the `ManagedClusterAddOn` would be like:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: helloworld
+  namespace: cluster1
+spec:
+  installNamespace: open-cluster-management-agent-addon
+  installVersion: v0
+Status:
+…
+  lastVersion: v1
+  currentVersion: unknown
+  conditions:
+  - lastTransitionTime: "2022-11-02T05:48:12Z"
+    message: could not rollback to invalid version v0
+    reason: Failed
+    status: "Flase"
+    type: Progressing
+```
+
+### Test Plan
+- All implementations will have thorough coverage of unit tests.
+- Integration tests will cover the user cases.
+
+### Graduation Criteria
+#### Alpha
+At first, This proposal will be in the alpha stage and needs to meet
+1. The new APIs are reviewed and accepted;
+2. Implementation is completed to support the functionalities;
+3. Develop test cases to demonstrate this proposal works correctly;
+
+#### Beta
+1. Need to revisit the API shape before upgrading to beta based on user feedback.
+
+### Upgrade / Downgrade Strategy
+TBD
+
+### Version Skew Strategy
+N/A
+
+## Alternatives
```

Comment:
Can you compare this against an approach that uses `customizedVariables`?  I can appreciate version as a special thing, but the general idea of a rolling deployment is generic to any configuration change.

Reply:
this is a great point :)

Reply:
Thanks for this comment, would provide an alternative solution.